### PR TITLE
UCT/API: make uct_iface_open backward compatible

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
make uct_iface_open backward compatible

## How ?
The same way as UCP, adding params argument with field_mask to API routines.

#3115

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8568/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8572/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8573/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8575/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8634/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp @bbenton can you pls take a look?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8646/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8648/ for details (Mellanox internal link).

File: src/uct/api/uct.h [hunk 1]
```diff
@@ -363,14 +363,69 @@ enum uct_cb_flags {
...
+   /**< Interface is opened on a specific device */
+   UCT_IFACE_OPEN_MODE_DEVICE          = UCS_BIT(0),
+
+   /**< Interface is opened on a specific address on the server side. This mode
+        is going to be deprecated in near future for a better API. */
+   UCT_IFACE_OPEN_MODE_SOCKADDR_SERVER = UCS_BIT(1),
+
+   /**< Interface is opened on a specific address on the client side This mode
+        is going to be deprecated in near future for a better API. */
+   UCT_IFACE_OPEN_MODE_SOCKADDR_CLIENT = UCS_BIT(2)
```
Comment (bbenton) [inline src/uct/api/uct.h]: Remove the '<' marker in the comment blocks since they come before, rather than after, the enum member.

**Comment by evgeny-leksikov** [comment]
@bbenton thank you for review! your comment is addressed.
@shamisp pls take a look.

**Comment by shamisp** [comment]
I understand motivation behind the changes, but to we have a plan for fixing OpenMPI UCT BTL ?

**Comment by shamisp** [comment]
@tonycurtis please take a look as well. thanks !

**Comment by evgeny-leksikov** [comment]
@hjelmn this PR and #3115 are start for making UCT backward compatible as you requested, WDT if we break it for better future?

**Comment by evgeny-leksikov** [comment]
@tonycurtis 
< In src/tools/info/tl_info.c, do we care about the string buffers having magic number sizes, e.g. lines 79, 99, 131?

It's not related to the API changes, I tried to do minimal changes to related to the API to pass CI tests and for easier review.

**Comment by tonycurtis** [comment]
@evgeny-leksikov fair enough...

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8691/ for details (Mellanox internal link).

**Comment by hjelmn** [comment]
Don't have much free time available this week but I will try to take a look tonight. If I don't get to it tonight it will be Saturday at the earliest.

**Pull request merged**